src/apyfixed.cc

```diff
@@ -995,6 +995,45 @@ APyFixed APyFixed::from_double(
     return result;
 }
 
+APyFixed APyFixed::from_unspecified_double(double value)
+{
+    if (std::isnan(value) || std::isinf(value)) {
+        throw std::domain_error(fmt::format("Cannot convert {} to fixed-point", value));
+    }
+
+    int exp = exp_of_double(value);
+    uint64_t man = man_of_double(value);
+
+    // Append mantissa hidden one
```

Comment:
Look at the new `APyFixed.set_from_double()` for properly handling subnormals.

Or better still, let this function only evaluate the necessary width of the resulting APyFixed, and reuse `APyFixed.set_from_double()` for the data casting.